### PR TITLE
tests: import remote tools

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This change introduces the new tools for remote commands from the snapd-testing-tools project

This change is needed to simplify the: tests: new remote tools - part 1 #11954

### CODE REVIEW

File: tests/lib/external/snapd-testing-tools/tools/tests.systemd [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    if [ -z "$start_line" ]; then
+        echo "tests.systemd: unit command line cannot be empty"
+        return 1
+    fi
+    if [ -f "/etc/systemd/system/$name.service" ]; then
+        echo "tests.systemd: unit service file already exist"
+        return 1  
+    fi
+
+    printf '[Unit]\nDescription=Support for test %s\n[Service]\nType=simple\nExecStart=%s\n[Install]\nWantedBy=multi-user.target\n' "unknown" "$start_line" > "/etc/systemd/system/$name.service"
```
Comment (mardy) [inline tests/lib/external/snapd-testing-tools/tools/tests.systemd:32]: Instead of hardcoding `"unknown"`, maybe we can use `$name` here?

**Review by mardy — COMMENTED** [review]
Nothing critical, just a couple of suggestions.

File: tests/lib/external/snapd-testing-tools/remote/remote.wait-for [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+        wait_for_reconnect_ssh
+        return
+    fi
+
+    while [ $attempts -ge 0 ]; do
+        echo -n '.'
+        attempts=$(( attempts - 1 ))
+        # The get_boot_id could fail because the connection is broken due to the reboot
+        last_boot_id="$(get_boot_id)" || true
+        if [[ "$last_boot_id" =~ .*-.*-.*-.*-.* ]] && [ "$last_boot_id" != "$initial_boot_id" ]; then
```
[the remark this answers is not in the document]
Comment (sergiocazzolato) [reply tests/lib/external/snapd-testing-tools/remote/remote.wait-for:116]: done

File: tests/lib/external/snapd-testing-tools/tools/tests.systemd (hunk 1 above)
Comment (sergiocazzolato) [reply tests/lib/external/snapd-testing-tools/tools/tests.systemd:32]: I think so, thanks for the suggestion

File: tests/lib/external/snapd-testing-tools/remote/remote.refresh [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+
+    echo "remote.refresh: starting kernel refresh process"
+    snap_name="$(remote.exec "snap list" | grep -E '(kernel$|kernel,)' | awk '{ print $1 }')"
+    if [ -z "$snap_name" ]; then
+        echo "remote.refresh: no kernel snap to update"
+        return
+    fi
+
+    process_refresh "$snap_name" "$refresh_channel"
+}
```
[the remark this answers is not in the document]
Comment (sergiocazzolato) [reply tests/lib/external/snapd-testing-tools/remote/remote.refresh]: @mardy I improved the code to reuse that part in all the refresh functions. Thanks for the review

**Review by mardy — APPROVED** [review]
Thanks for the updates, LGTM!

**Pull request merged**